smartcontract/service/wasmvm/storage.go

```diff
@@ -25,12 +25,23 @@ import (
 	"github.com/ontio/ontology/core/states"
 )
 
-func StorageRead(proc *exec.Process, keyPtr uint32, klen uint32, val uint32, vlen uint32, offset uint32) uint32 {
+func readVmMem(proc *exec.Process, ptr uint32, len uint32) ([]byte, error) {
+	if uint64(proc.MemSize()) < uint64(ptr)+uint64(len) {
+		return nil, errors.New("contract create len is greater than memory size")
+	}
+	keybytes := make([]byte, len)
+	_, err := proc.ReadAt(keybytes, int64(ptr))
+	if err != nil {
+		return nil, err
+	}
 
+	return keybytes, nil
+}
```

Comment:
this function put it to runtime.go or wasm_service.go is better

Reply:
extract out to utils.go